### PR TITLE
Use upstream RPC-MaaS (newton)

### PR DESCRIPTION
**Pull request by cloudnull**
This change converts our deployment of RPC MaaS from the old role method
to the new playbook method. This change will provide us access to a
faster and more flexible monitoring solution and allow all customers to
benifit from Monitoring as a Service using a unified code base.

The maas group_vars have been removed and the values of the old file
have been placed into the `user_osa_variables_defaults.yml` file. This
has been done to allow the rpc-maas playbooks to inherit from the user
variables in cases where the group vars not not available.

Signed-off-by: Kevin Carter <kevin.carter@rackspace.com>
(cherry picked from commit 792950d24e7e8dab8d14b44524dd2717265dca1c)

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_artifact_all

**Comment by odyssey4me** [comment]
recheck_cit_artifact_trusty_Apt
recheck_cit_artifact_trusty_Container
recheck_cit_artifact_trusty_Git
recheck_cit_artifact_trusty_Python
recheck_cit_artifact_xenial_Apt
recheck_cit_artifact_xenial_Container
recheck_cit_artifact_xenial_Git
recheck_cit_artifact_xenial_Python

**Comment by odyssey4me** [comment]
recheck_cit_trusty_deploy_ceph

**Comment by odyssey4me** [comment]
I think this may need a rebase to bring in the recent fixes that merged.

File: scripts/bootstrap-ansible.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ source ${BASE_DIR}/scripts/functions.sh
 ## Vars ----------------------------------------------------------------------
 
 # Set the role fetch mode to any option [galaxy, git-clone]
-export ANSIBLE_ROLE_FETCH_MODE=${ANSIBLE_ROLE_FETCH_MODE:-galaxy}
+export ANSIBLE_ROLE_FETCH_MODE=${ANSIBLE_ROLE_FETCH_MODE:-git-clone}
```
Comment (evrardjp) [inline scripts/bootstrap-ansible.sh]: Why is this needed?

File: scripts/deploy-rpc-playbooks.sh [hunk 2]
```diff
@@ -38,11 +38,13 @@ fi
 
 # deploy and configure RAX MaaS
 if [[ "${DEPLOY_MAAS}" == "yes" ]]; then
+  # Download the latest release of rpc-maas
+  run_ansible maas-get.yml
```
Comment (evrardjp) [inline scripts/deploy-rpc-playbooks.sh]: Could you include the same line into scripts/leapfrog/post_leap.sh please?

**Comment by evrardjp** [comment]
@hughsaunders The change here include a backport of #2377 fix.

File: scripts/deploy-rpc-playbooks.sh (hunk 2 above)
Comment (cloudnull) [inline scripts/deploy-rpc-playbooks.sh]: done

File: scripts/bootstrap-ansible.sh (hunk 1 above)
Comment (cloudnull) [inline scripts/bootstrap-ansible.sh]: reverted

**Comment by evrardjp** [comment]
Looks good. The line ``Signed-off-by: Kevin Carter <kevin.carter@rackspace.com>`` appears twice, but that's detail.

I'm waiting to see the leapfrog job results.

**Comment by evrardjp** [comment]
Good for me if it passes leapfrog gating.

**Comment by cloudnull** [comment]
recheck_cit_artifact_trusty_Apt
recheck_cit_artifact_xenial_Apt

**Comment by odyssey4me** [comment]
The leapfrog failed with:

```
TASK [Clone RPC-MaaS] **********************************************************
[0;31mfatal: [localhost]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'maas_release' is undefined\n\nThe error appears to have been in '/opt/rpc-openstack/rpcd/playbooks/maas-get.yml': line 20, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  tasks:\n    - name: Clone RPC-MaaS\n      ^ here\n"}[0m
```

**Comment by cloudnull** [comment]
recheck_cit_trusty_leapfrogupgrade_swift

**Comment by evrardjp** [comment]
recheck_cit_trusty_deploy_swift
recheck_cit_trusty_deploy_ceph
recheck_cit_trusty_leapfrogupgrade_swift
recheck_cit_xenial_deploy_swift

**Comment by evrardjp** [comment]
I'm babysitting this one

File: scripts/leapfrog/post_leap.sh [hunk 3]
```diff
@@ -19,12 +19,20 @@
 set -e -u -x
 set -o pipefail
 
+export RPCM_VARIABLES=${RPCM_VARIABLES:-/etc/openstack_deploy/user_rpcm_variables.yml}
+
 echo "POST LEAP STEPS"
+
+if [[ ! -f "${RPCM_VARIABLES}" ]]; then
+  cp "${RPCO_DEFAULT_FOLDER}/etc/openstack_deploy/user_rpcm_variables.yml" "${RPCM_VARIABLES}"
```
Comment (evrardjp) [inline scripts/leapfrog/post_leap.sh]: cp line is wrong, and should probably be 👍 
``
cp "${RPCO_DEFAULT_FOLDER}/rpcd/etc/openstack_deploy/user_rpcm_variables.yml" "${RPCM_VARIABLES}"
``

**Comment by evrardjp** [comment]
Also:
```
TASK [Drop .raxrc file] ********************************************************
fatal: [rantdsp-57-27f1]: FAILED! => {"changed": false, "failed": true, "msg": "AnsibleUndefinedVariable: {{ rackspace_cloud_auth_url }}: 'rackspace_cloud_auth_url' is undefined"}
```

**Comment by evrardjp** [comment]
I will fix that ASAP.

**Comment by cloudnull** [comment]
recheck_cit_trusty_deploy_ceph
